src/backend/distributed/utils/node_metadata.c

```diff
@@ -895,6 +896,31 @@ FindWorkerNodeAnyCluster(const char *nodeName, int32 nodePort)
 }
 
 
+/*
+ * GetWorkerNodeByNodeId iterates over worker nodes and returns a pointer to the
+ * one that has the given id
+ */
+WorkerNode *
+GetWorkerNodeByNodeId(int nodeId)
```

Comment:
There seems to be a better alternative to this function here: https://github.com/citusdata/citus/blob/eb35743c3fd937e1b96c1ee39979aa9dead6d222/src/backend/distributed/utils/metadata_cache.c#L578-L583

The main difference is I used the index on `pg_dist_node`, whereas this one uses `WorkerNodeArray` that is already in a global variable in memory.

I guess I better remove my function and use this one here